### ISSUE TITLE
Types references in the FixedBufferAttribute do not use standard desktop unification for identity comparison

### ISSUE THREAD

**Issue by agocke** [dotnet/roslyn#2385]
An example test to demonstrate the issue:

```csharp
        [Fact]
        public void FixedSizeBuffer()
        {
            var text1 = @"
using System;
using System.Runtime.InteropServices;

public static class R
{
    public unsafe struct S
    {
        public fixed byte Buffer[16];
    }
}";
            var comp1 = CreateCompilation(text1, references: new[] { MscorlibRef_v20 },
                options: TestOptions.UnsafeDebugDll);

            var ref1 = comp1.EmitToImageReference();

            var text2 = @"
using System;

class C
{
    unsafe void M(byte* p)
    {
        R.S* p2 = (R.S*)p;
        IntPtr p3 = M2((IntPtr)p2[0].Buffer);
    }

    unsafe IntPtr M2(IntPtr p) => p;
}";
            var comp2 = CreateCompilationWithMscorlib45(text2,
                references: new[] { ref1 },
                options: TestOptions.UnsafeDebugDll);
            comp2.VerifyDiagnostics();
        }
```

This should not produce an error as the emitted type in the first assembly will be `System.Byte` from `mscorlib, Version=2.0.0.0...` and the second compilation references `mscorlib, Version 4.5.0.0...`, which means that `System.Byte` should be unified to the v4.5 type. However, `PEFieldSymbol` doesn't use the appropriate unification algorithm, so it produces the following error:

```csharp
    // (9,24): error CS0030: Cannot convert type 'R.S.<Buffer>e__FixedBuffer' to 'System.IntPtr'
    //         IntPtr p3 = M2((IntPtr)p2[0].Buffer);
    Diagnostic(ErrorCode.ERR_NoExplicitConv, "(IntPtr)p2[0].Buffer").WithArguments("R.S.<Buffer>e__FixedBuffer", "System.IntPtr").WithLocation(9, 24)
```

**Comment by agocke** [comment]
@tmat FYI

**Comment by gafter** [comment]
@tmat Can you please Verify this?
